CRM/Case/BAO/Case.php

```diff
@@ -1217,24 +1209,52 @@ public static function getCaseActivity($caseID, &$params, $contactID, $context =
       // if there are file attachments we will return how many and, if only one, add a link to it
       if (!empty($dao->attachment_ids)) {
         $attachmentIDs = array_unique(explode(',', $dao->attachment_ids));
-        $caseActivity['no_attachments'] = count($attachmentIDs);
+        $caseActivities[$caseActivityId]['no_attachments'] = count($attachmentIDs);
         $url .= implode(' ', CRM_Core_BAO_File::paperIconAttachment('civicrm_activity', $caseActivityId));
       }
 
-      $caseActivity['links'] = $url;
-
-      array_push($caseActivities, $caseActivity);
+      $caseActivities[$caseActivityId]['links'] = $url;
     }
     $dao->free();
 
     $caseActivitiesDT = array();
-    $caseActivitiesDT['data'] = $caseActivities;
+    $caseActivitiesDT['data'] = array_values($caseActivities);
     $caseActivitiesDT['recordsTotal'] = $caseCount;
     $caseActivitiesDT['recordsFiltered'] = $caseCount;
 
     return $caseActivitiesDT;
   }
 
+  /**
+   * Helper function to generate a formatted contact link/name for display in the Case activities tab
+   *
+   * @param $contactId
+   * @param $contactName
+   *
+   * @return string
+   */
+  private static function formatContactLink($contactId, $contactName) {
+    if (!isset(Civi::$statics[__CLASS__]['contactViewUrl'])) {
+      Civi::$statics[__CLASS__]['contactViewUrl'] = CRM_Utils_System::url("civicrm/contact/view", "reset=1&cid=", FALSE, NULL, FALSE);
+    }
+    $contactViewUrl = Civi::$statics[__CLASS__]['contactViewUrl'];
```

Comment:
This static cache of composed url strings looks to me like a microoptimization that is happening on the wrong level. If we're going to do it at all it should be a patch to `CRM_Utils_System::url`. But frankly I'm doubtful that it is useful. The only system that triggers a database hit for is Drupal, and Drupal has its own performance caching system.